File: main/plugins/org.talend.camel.designer/src/main/java/org/talend/camel/designer/util/CamelSpringUtil.scala
package org.talend.camel.designer.util

import org.talend.camel.core.model.camelProperties.CamelProcessItem
import java.util.Scanner

object CamelSpringUtil {

  var TMP_SPRING_CONTENT: Option[String] = None

  def getDefaultContent(item: CamelProcessItem): String = {
    if (TMP_SPRING_CONTENT.isEmpty) {
      TMP_SPRING_CONTENT = Some(getTmpSpringContent())
    }
    TMP_SPRING_CONTENT.get.replace("${RouteItemName}", item.getProperty().getLabel())
  }

  def getTmpSpringContent(): String = {
    try {
      val is = this.getClass().getResourceAsStream("spring.xml")
      val defaultContent = new Scanner(is).useDelimiter("\\A").next()
      is.close
      defaultContent
    } catch {
      case e: Exception =>
        e.printStackTrace()
        ""
    }
  }
}

File: main/plugins/org.talend.camel.designer/src/main/java/org/talend/camel/designer/util/CamelSpringUtil.java
package org.talend.camel.designer.util;

import java.io.InputStream;
import org.apache.commons.io.IOUtils;
import org.talend.camel.core.model.camelProperties.CamelProcessItem;

public class CamelSpringUtil {

	private static String TMP_SPRING_CONTENT = null;

	private CamelSpringUtil() {
	}

	public static String getDefaultContent(CamelProcessItem item) {
		if (TMP_SPRING_CONTENT == null) {
			TMP_SPRING_CONTENT = getTmpSpringContent();
		}

		return TMP_SPRING_CONTENT.replace("${RouteItemName}", item.getProperty().getLabel());
	}

	private static String getTmpSpringContent() {
		InputStream is = null;
		try {
			is = CamelSpringUtil.class.getResourceAsStream("spring.xml");
			return IOUtils.toString(is);
		} catch (Exception e) {
			e.printStackTrace();
			return "";
		} finally {
			IOUtils.closeQuietly(is);
		}
	}
}
